code/modules/antagonists/traitor/contractor/datums/syndicate_contract.dm

```diff
@@ -293,9 +319,9 @@
   * * P - The extraction portal.
   */
 /datum/syndicate_contract/proc/target_received(mob/living/M, obj/effect/portal/redspace/contractor/P)
+	INVOKE_ASYNC(src, .proc/clean_up)
```

Comment:
Why is this called in async?

Reply:
Because `handle_target_experience` calls `sleep` which pauses the proc. `clean_up` was originally called directly after and would only actually happen when all `sleep`s were done. It's asynchronous because there is some behavior tied to the portal in `handle_target_experience` prior to its removal.